lib/services/static-files-api-service.js

```diff
@@ -0,0 +1,155 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var fs = require('fs');
+var path = require('path');
+var walking = require('walk');
+
+module.exports = staticFilesApiServiceFactory;
+
+di.annotate(staticFilesApiServiceFactory, new di.Provide('Http.Services.Api.StaticFiles'));
+di.annotate(staticFilesApiServiceFactory,
+    new di.Inject(
+      'Constants',
+      'Logger',
+      'Promise',
+      'Services.Configuration',
+      'Services.Waterline'
+    )
+);
+
+
+function staticFilesApiServiceFactory(
+    constants,
+    Logger,
+    Promise,
+    configuration,
+    waterline
+) {
+    var logger = Logger.initialize(staticFilesApiServiceFactory);
+    var skuPath = constants.HttpStaticDir.skupack;
+    var staticPath = constants.HttpStaticDir.systemDefault;
+    var userPath = constants.HttpStaticDir.userDefault;
+    var walker;
+    var walk = Promise.promisifyAll(walking);
+    Promise.promisifyAll(fs);
+
+    function StaticFilesApiService() {
+    }
+
+
+    StaticFilesApiService.prototype.pairSkupackIds = function(dirs) {
+        return Promise.try(function() {
+            return fs.readdirAsync(skuPath).filter(function(item) {
+                return (fs.lstatAsync(path.join(skuPath, item)))
+                .then(function(stat) {
+                    return stat.isDirectory();
+                }); 
+            }).reduce(function(result, item) {
+                return Promise.try(function() {
+                    return waterline.skus.findOne({id : item})
+                    .then(function(sku) {
+                        result[item] = sku.name;
+                        return result;
+                    });
+                })
+                .catch(function(e) {
+                    logger.debug("Directory " + skuPath + "/" + item + " is not in the database",
+                        e);
+                    return result;
+                });
+            },{}).then(function(pairs) {
+                dirs.push(skuPath);
+                return pairs;
+            });
+        })
+        .catch(function(e) {
+            logger.debug("StaticFilesApiService.pairSkupackIds", e);
+            return {};
+        });
+    };
+
+
+    StaticFilesApiService.prototype.walkDirectory = function(dir) {
+        return Promise.try(function() {
+            return new Promise(function(resolve, reject) {
+                var walkResults = [];
+                walker = walk.walk(dir, {followLinks: false});
+
+                walker.on("file", function(root, fileStats, next) {
+                    Promise.resolve().then(function() {
```

Comment:
@lacarb : the sub directories may have already been covered, if your testing is OK, that's OK.

But for this promise itself, Since the code didn't listen on "error" event, so you want to intentionally ignore all the walk error? But some error cannot be ignored, for example, for a wrong directory input, the walk should send some error and you will get empty result during `end` event.